### PR TITLE
Roles-change-requests

### PR DESCRIPTION
**Pull request by devo-wm**
https://whitemoondreams.atlassian.net/browse/TC-3472
> Ability to create a new custom role (done in previous pr)
> Ability to delete a custom role (Default roles are locked from deletion)
> Ability to assign permissions to a custom role (excludes any site owner specific permissions)
> Ability to edit permissions of default supplied roles (Site owner is unavailable to edit)

 - Adds delete confirmation modal
 - Blocks deletion of default roles
 - Blocks deletion of locked roles
 - Fixes repeated permissions being listed in the detail view.
 - Fixes site filtering query
 - Fixes issue where newly created custom role wasn't showing up in the detail and manage view sidebar
 - Updates search/list endpoint for custom modal view to use the same url and view
 - Minor update to bootstrap templating.

### CODE REVIEW

**Review by rhimmelbauer — APPROVED** [review]
Nice work it looks very good. Just don't forget to run translations `./manage.py makemessages -l ja`

File: permafrost/templates/permafrost/permissions_modal.html [hunk 1]
```diff
@@ -31,7 +31,7 @@ <h2 class="text-primary-grey mb-3">
             </div>
         </div>
         <form method="post" action="{% url 'permafrost:custom-role-add-permissions' slug=object.slug %}" id="perm-modal-form">
-            <input type="hidden" name="csrfmiddlewaretoken" value="{{ csrf_token }}">
+            {% csrf_token %}
```
Comment (fahzee1) [inline permafrost/templates/permafrost/permissions_modal.html:34]: For some reason this wasnt working for me. Thanks for confirming it works

File: permafrost/views.py [hunk 2]
```diff
@@ -145,9 +146,19 @@ def handle_no_permission(self):
 class FilterByRequestSiteQuerysetMixin:
     def get_queryset(self):
         if hasattr(self.request, 'site'):
-            return PermafrostRole.objects.filter(site=self.request.site)
+            return PermafrostRole.objects.filter(site=self.request.site, deleted=False)
         return super().get_queryset()
 
+class GetRoleExternalPermissionsMixin:
+    def get_perms_excluding_current_role(self, context):
```
Comment (fahzee1) [inline permafrost/views.py:153]: nice

**Review by fahzee1 — APPROVED** [review]
Looks good!

**Pull request merged**